datafusion/src/physical_optimizer/pruning.rs

```diff
@@ -748,4 +821,40 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn prune_api() {
```

Comment:
This shows end-to-end how to use the prune API (what I want to be able to do in IOx)